src/test/database/fakeListRemote.spec.ts

```diff
@@ -0,0 +1,101 @@
+import * as pathLib from "path";
+import * as chai from "chai";
+
+import { ListRemote } from "../../database/listRemote";
+
+const expect = chai.expect;
+
+/**
+ * `FakeListRemote` is a test fixture for verifying logic lives in the
+ * `DatabaseRemove` class. It is essentially a mock for the Realtime Database
+ * that accepts a JSON tree to serve upon construction.
+ */
+export class FakeListRemote implements ListRemote {
+  data: any;
+  delay: number;
+
+  /**
+   * @param data the fake database structure. Each leaf is an integer
+   *   representing the subtree's size.
+   */
+  constructor(data: any) {
+    this.data = data;
+    this.delay = 0;
+  }
+
+  async listPath(
+    path: string,
+    numChildren: number,
+    startAfter?: string,
+    timeout?: number
+  ): Promise<string[]> {
+    if (timeout === 0) {
+      throw new Error("timeout");
+    }
+    const d = this.dataAtPath(path);
+    if (d) {
+      let keys = Object.keys(d);
+      /*
+       * We mirror a critical implementation detail of here. Namely, the
+       * `startAfter` option (if it exists) is applied to the resulting key set
+       * before the `limitToFirst` option.
+       */
+      if (startAfter) {
+        keys = keys.filter((key) => key > startAfter);
+      }
+      keys = keys.slice(0, numChildren);
+      return keys;
+    }
+    return [];
```

Comment:
nit: can reduce complexity by saying `let keys = Object.keys(d || {})` (or by switching this to an early return)